SpatialGDK/Source/SpatialGDK/Private/Interop/CrossServerRPCSender.cpp

```diff
@@ -27,8 +27,8 @@ void CrossServerRPCSender::SendCommand(const FUnrealObjectRef InTargetObjectRef,
 
 	CommandRequest CommandRequest(SpatialConstants::SERVER_TO_SERVER_COMMAND_ENDPOINT_COMPONENT_ID,
 								  SpatialConstants::UNREAL_RPC_ENDPOINT_COMMAND_ID);
-
-	uint64 UniqueRPCId = FMath::RandHelper(INT_MAX);
+	FGuid Guid = FGuid::NewGuid();
+	uint64 UniqueRPCId = (((uint64)Guid.C) << 32) + (uint64)Guid.D;
```

Comment:
Worth considering using an actual high quality random number generator? PCG is pretty small?

Reply:
and really performant.

Reply:
This entire implementation is kind of stop-gap, as it will never actually be perfect. There is also a crypto lib currently in the engine that seems to do a pretty good random, but I didn't want to introduce the dependency just because of this.
This GUID call seems to resolve to some OS-internal function, but I can research a bit if it's high-quality enough for us.
(And tbf, the usage here - to generate ids for rpcs seems better suited to guids anyway, just concerned with throwing away the higher bits).

Reply:
I checked a bit - on windows it's something really opaque. As a fallback on an unknown platform it's mostly system time + cycles in the lower bits, so that is not great. But the actual implementation on unix is using urandom (or at least tries to).

Reply:
fair enough